civicrm_views_integration/civicrm_views_integration.install

```diff
@@ -2,41 +2,50 @@
 
 /**
  * Implements hook_install().
- * Appends settings.php file with prefixes for all current civicrm tables
- * when installing this module for the first time
  */
 function civicrm_views_integration_install() {
-  civicrm_initialize();
-
   global $databases;
-
   // Don't do anything if there are prefixes already defined
-  if (is_array($databases['default']['default']['prefix'])) {
-    return NULL;
+  if (!is_array($databases['default']['default']['prefix'])) {
+    // Appends settings.php file with prefixes for all current civicrm tables.
+    _civicrm_views_integrating_generate_db_prefixes();
   }
+}
 
-  $config = CRM_Core_Config::singleton();
-
-  // Only add the prefix if civicrm and drupal database are not the same one
-  if ($config->dsn != $config->userFrameworkDSN) {
-    $tableNames = CRM_Core_DAO::GetStorageValues(NULL, 0, 'Name');
-
-    $tablePrefixes = '$databases[\'default\'][\'default\'][\'prefix\']= array(';
-
-    // add default prefix: the drupal database prefix
-    $tablePrefixes .= "\n  'default' => '" .$databases['default']['default']['prefix'] . "',";
-
-    $dsnArray = DB::parseDSN($config->dsn);
-    $prefix = "`{$dsnArray['database']}`.";
+/**
+ * Implements hook_requirements().
```

Comment:
@omarabuhussein because hooks are meant to be generic and its generally not implemented for any specific task so according to Drupal standards its best to avoid writing descriptions to for them, as one can reuse the same hook for different functionality also.